agent/lib/kontena/network_adapters/weave.rb

```diff
@@ -25,7 +25,8 @@ def initialize(autostart = true)
       subscribe('agent:node_info', :on_node_info)
       subscribe('ipam:start', :on_ipam_start)
       async.ensure_images if autostart
-      #async.cleanup_ipam if autostart
+      Actor[:weave_executor] = WeaveExecutor.pool
```

Comment:
Good point. Celluloid terminates pools when garbage collected, but when it's in the Actor registry it will not get collected. As the executor pool is not used outside of `WeaveAdapter` I'll move it out of the registry and store as instance variable. That way it will also get cleaned via garbage collection.